tests/ga/test_update.py

```diff
@@ -1897,6 +1830,77 @@ def test_it_should_reset_legacy_blacklisted_agents_on_process_start(self):
+                                            with mock_update_handler(
+                                                    protocol,
+                                                    exthandlers_handler=mock_exthandlers_handler,
+                                                    remote_access_handler=mock_ra_handler,
+                                                    autoupdate_enabled=autoupdate_enabled,
+                                                    check_daemon_running=check_daemon_running
+                                            ) as update_handler:
+
+                                                if emit_restart_event is not None:
+                                                    update_handler._emit_restart_event = emit_restart_event
+
+                                                if isinstance(os.getppid, MagicMock):
+                                                    update_handler.run()
+                                                else:
+                                                    with patch('os.getppid', return_value=42):
+                                                        update_handler.run()
+
+                                                self.assertEqual(1, mock_monitor.call_count)
+                                                self.assertEqual(1, mock_env.call_count)
+                                                self.assertEqual(1, mock_collect_logs.call_count)
+                                                self.assertEqual(1, mock_telemetry_send_events.call_count)
+                                                self.assertEqual(1, mock_event_collector.call_count)
+                                                self.assertEqual(expected_exit_code, update_handler.get_exit_code())
+
+                                                if update_handler.get_iterations_completed() > 0:  # some test cases exit before executing extensions or remote access
+                                                    self.assertEqual(1, mock_exthandlers_handler.run.call_count)
+                                                    self.assertEqual(1, mock_ra_handler.run.call_count)
+
+                                                return update_handler
+
+    def test_run(self):
+        self._test_run()
+
+    def test_run_stops_if_update_available(self):
+        with patch('azurelinuxagent.ga.update.UpdateHandler._download_agent_if_upgrade_available', return_value=True):
+            update_handler = self._test_run(autoupdate_enabled=True)
+            self.assertEqual(0, update_handler.get_iterations_completed())
+
+    def test_run_stops_if_orphaned(self):
+        with patch('os.getppid', return_value=1):
+            update_handler = self._test_run(check_daemon_running=True)
+            self.assertEqual(0, update_handler.get_iterations_completed())
+
+    def test_run_clears_sentinel_on_successful_exit(self):
+        update_handler = self._test_run()
```

Comment:
Should we create the sentinel file before starting the run? Doesn't look like this test is explicitly guaranteeing that `run` clears the sentinel file, just that it doesn't create one on a successful termination.

Reply:
the only intention of this PR is to migrate those tests to use the mock_update_handler. no other improvements (and these tests need quite a few) were made